boto/dynamodb2/table.py

```diff
@@ -379,17 +379,22 @@ def describe(self):
         # This is leaky.
         return result
 
-    def update(self, throughput, global_indexes=None):
+    def update(self, throughput=None, global_indexes=None):
```

Comment:
The responsibility of this method is to update table attributes. 

Here, the addition allows the creation, update and deletion of GSI. Then it seems like those changes belong in different (potentially smaller) methods. It will keep the complexity of this method to the minimum.

What do you think? @danielgtaylor